app/react/Layout/TableViewer.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { Component } from 'react';
+
+import { connect } from 'react-redux';
+import { TableRow } from 'app/Library/components/TableRow';
+
+export interface DocumentViewerProps {
+  rowListZoomLevel: number;
+  documents: any;
+  storeKey: 'library' | 'uploads';
+  clickOnDocument: (...args: any[]) => any;
+  onSnippetClick: (...args: any[]) => any;
+  deleteConnection: (...args: any[]) => any;
+  search: any;
+  templates: any;
+  thesauris: any;
+  columns: any[];
+}
```

Comment:
The amount of "any"s here, looks like we are just using TS for the sake of using it, I don't think that is the correct approach.

But there is a lot of problems here: templates and thesauris are brought directly from the state (store), they are not changed, and they are passed on to TableRow.  That is completely unneeded.  You can just mapStateToProps in TableRow to those two values and remove that from here.

I understand that is how refactors start, but we need to take them to the next step.

"Search" is not being used in this component, at all!

Reply:
Totally agree! That's exactly what happened: this refactor took many many shapes on the process, and we failed to revisit some initial steps. I'm glad you caught this things!